src/core/sock/sockinfo.h

```diff
@@ -328,6 +329,8 @@ class sockinfo {
     inline bool set_flow_tag(uint32_t flow_tag_id);
     inline void sock_pop_descs_rx_ready(descq_t *cache);
 
+    bool has_stats() const { return m_has_stats; }
```

Comment:
[question] Have you considered --disable-socket-stats option will converts `has_stats()` to false in compile time? This will optimize out a lot of code.

Reply:
Yes. i considered it. I even tried to comment out all the stats in DP and didn't see any change in IOPs. I decided to keep it this way to avoid another compile time flag.